lib/runner.js

```diff
@@ -449,7 +450,24 @@ Runner.prototype.runTests = function(suite, fn){
     // grep
     var match = self._grep.test(test.fullTitle());
     if (self._invert) match = !match;
-    if (!match) return next();
+    if (!match) {
+        // Run immediately only if we have defined a grep. When we
+        // define a grep — It can cause maximum callstack error if
+        // the grep is doing a large recursive loop by neglecting
+        // all tests. The run immediately function also comes with
+        // a performance cost. So we don't want to run immediately
+        // if we run the whole test suite, because running the whole
+        // test suite don't do any immediate recursive loops. Thus,
+        // allowing a JS runtime to breathe.
+        if (self._grep !== self._defaultGrep) {
+            Runner.immediately(function() {
+                next();
+            });
+        } else {
+            next();
+        }
+        return;
```

Comment:
I don't understand your question. Though I need to a return statement here because I don't want to execute the rest of the function body.